iocell/src/main/scala/chisel/IOCell.scala

```diff
@@ -143,12 +147,47 @@ object IOCell {
    * @param anaFn A function to generate an AnalogIOCell to use for analog signals
    * @return A Seq of all generated IO cell instances
    */
-  def generateFromSignal[T <: Data](coreSignal: T, padSignal: T, name: Option[String] = None,
+  val toSyncReset: (Reset) => Bool = _.toBool
+  val toAsyncReset: (Reset) => AsyncReset = _.asAsyncReset
+  def generateFromSignal[T <: Data, R <: Reset](
+    coreSignal: T,
+    padSignal: T,
+    name: Option[String] = None,
     inFn: () => DigitalInIOCell = IOCell.genericInput,
     outFn: () => DigitalOutIOCell = IOCell.genericOutput,
-    anaFn: () => AnalogIOCell = IOCell.genericAnalog): Seq[IOCell] =
+    anaFn: () => AnalogIOCell = IOCell.genericAnalog,
+    concretizeResetFn : (Reset) => R = toSyncReset): Seq[IOCell] =
```

Comment:
Is there a reason to do this here rather than mirror the `abstractResetAsAsync` argument from  `generateIOFromSignal`? Just curious why the arguments are slightly different.

Reply:
I wrote this code first and i wanted to mirror the argument list above, before realizing that user probably only cares about two cases. I'll make it consistent in using the bool. It simplifies the type signature of the function too.